plugins/woocommerce/src/Blocks/Assets/AssetDataRegistry.php

```diff
@@ -297,24 +297,15 @@ public function exists( $key ) {
 	 * @param string  $key              The key used to reference the data being registered. This should use camelCase.
 	 * @param mixed   $data             If not a function, registered to the registry as is. If a function, then the
 	 *                                  callback is invoked right before output to the screen.
-	 * @param boolean $check_key_exists If set to true, duplicate data will be ignored if the key exists.
+	 * @param boolean $check_key_exists Deprecated. If set to true, duplicate data will be ignored if the key exists.
 	 *                                  If false, duplicate data will cause an exception.
-	 *
-	 * @throws InvalidArgumentException  Only throws when site is in debug mode. Always logs the error.
 	 */
 	public function add( $key, $data, $check_key_exists = false ) {
-		if ( $check_key_exists && $this->exists( $key ) ) {
-			return;
-		}
-		try {
-			$this->add_data( $key, $data );
-		} catch ( Exception $e ) {
-			if ( $this->debug() ) {
-				// bubble up.
-				throw $e;
-			}
-			wc_caught_exception( $e, __METHOD__, [ $key, $data ] );
+		if ( $check_key_exists ) {
```

Comment:
For this since we're deprecating the parameter, what do you think if we check for `isset` instead. This way, the caller will get a deprecated feedback whether they passed 'true' or 'false' there.

Reply:
Sure, if you don't mind please create a separate issue for it.

Reply:
Thanks, absolutely! https://github.com/woocommerce/woocommerce/issues/46349